cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java

```diff
@@ -113,8 +113,17 @@ public void create(HttpRequest request, HttpResponder responder, @PathParam("nam
       }
 
       NamespaceConfig config = metadata.getConfig();
-      if (config != null && !Strings.isNullOrEmpty(config.getSchedulerQueueName())) {
-        builder.setSchedulerQueueName(config.getSchedulerQueueName());
+      if (config != null) {
+        if (!Strings.isNullOrEmpty(config.getSchedulerQueueName())) {
+          builder.setSchedulerQueueName(config.getSchedulerQueueName());
+        }
+        // both or none of the following two must be set
```

Comment:
its not enforced or required currently; removed the comment.
It was just a potential requirement we could have

On Tue, Jul 5, 2016 at 1:32 PM, Bhooshan Mogal <notifications@github.com>
wrote:

> In
> cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java
> <https://github.com/caskdata/cdap/pull/5975#discussion_r69632762>:
>
> > @@ -113,8 +113,17 @@ public void create(HttpRequest request, HttpResponder responder, @PathParam("nam
> >        }
> >
> >        NamespaceConfig config = metadata.getConfig();
> > -      if (config != null && !Strings.isNullOrEmpty(config.getSchedulerQueueName())) {
> > -        builder.setSchedulerQueueName(config.getSchedulerQueueName());
> > +      if (config != null) {
> > +        if (!Strings.isNullOrEmpty(config.getSchedulerQueueName())) {
> > +          builder.setSchedulerQueueName(config.getSchedulerQueueName());
> > +        }
> > +        // both or none of the following two must be set
>
> do we enforce this constraint? if so, it seems like the if condition on
> line 124 should be nested inside the if condition on line 121?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/caskdata/cdap/pull/5975/files/2b51590a6d3be9ad56fdd805040db06034e37800#r69632762>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACU_ETXilcQ_9sgQFKLXi61HcGMXSwmSks5qSr9ygaJpZM4JFe4X>
> .
>